content/docs/1.5.0/snapshots-and-backups/csi-snapshot-support/csi-volume-snapshot-associated-with-longhorn-backing-image.md

```diff
@@ -0,0 +1,225 @@
+            name: test-on-demand-backing
+        spec:
+            volumeSnapshotClassName: longhorn-snapshot-vsc
+            driver: driver.longhorn.io
+            deletionPolicy: Delete
+            source:
+              # NOTE: change this to provide the correct parameters
+              snapshotHandle: bi://backing?backingImageDataSourceType=download&backingImage=test-bi&url=https%3A%2F%2Flonghorn-backing-image.s3-us-west-1.amazonaws.com%2Fparrot.qcow2&backingImageChecksum=bd79ab9e6d45abf4f3f0adf552a868074dd235c4698ce7258d521160e0ad79ffe555b94e7d4007add6e1a25f4526885eb25c53ce38f7d344dd4925b9f2cb5d3b
+        volumeSnapshotRef:
+            name: test-snapshot-on-demand-backing
+            namespace: default
+        ```
+
+2. `export-from-volume`: Users need to provide following parameters
+    - `backingImageDataSourceType`: `export-form-volume` for on-demand export.
+    - `backingImage`: Name of the BackingImage
+    - `volume-name`: Volume to be exported for the BackingImage
+    - `export-type`: Currently Longhorn supports `raw` or `qcow2`
+    - example yaml:
+        ```yaml
+        apiVersion: snapshot.storage.k8s.io/v1
+        kind: VolumeSnapshotContent
+        metadata:
+        name: test-on-demand-backing
+        spec:
+        volumeSnapshotClassName: longhorn-snapshot-vsc
+        driver: driver.longhorn.io
+        deletionPolicy: Delete
+        source: 
+          # NOTE: change this to provide the correct parameters
+          snapshotHandle: bi://backing?backingImageDataSourceType=export-from-volume&backingImage=test-bi&volume-name=vol-export-src&export-type=qcow2
+        volumeSnapshotRef:
+            name: test-snapshot-on-demand-backing
+            namespace: default
+        ```
+
+Create the associated `VolumeSnapshot` object with the `name` field set to `test-snapshot-on-demand-backing`, where the `source` field refers to a `VolumeSnapshotContent` object via the `volumeSnapshotContentName` field.
+
+This differs from the creation of a BackingImage, in which case the `source` field refers to a `PerstistentVolumeClaim` via the `persistentVolumeClaimName` field.
+
+Only one type of reference can be set for a `VolumeSnapshot` object.
+
+```yaml
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshot
+metadata:
+  name: test-snapshot-on-demand-backing
+spec:
+  volumeSnapshotClassName: longhorn-snapshot-vsc
+  source:
+    volumeSnapshotContentName: test-on-demand-backing
+```
+
+Now you can create a `PerstistantVolumeClaim` object that refers to the newly created `VolumeSnapshot` object.
+Longhorn will create the BackingImage with the parameters provide in the `snapshotHandle`.
+For an example see [Restore PVC from CSI VolumeSnapshot Associated With Longhorn BackingImage](#restore-pvc-from-csi-volumesnapshot-associated-with-longhorn-backingimage) above.
+
+
+## Current Limitation
+* Longhorn volume must be in the attached state to create/restore BackingImage using CSI VolumeSnapshot mechanism.
```

Comment:
Don't we have auto attach/detach to handle CSI snapshot? Is this limitation still valid?